src/bin/nydusd/fs_cache.rs

```diff
@@ -428,13 +436,16 @@ impl FsCacheHandler {
             Ok((blob, blob_size)) => {
                 let mut state = self.state.lock().unwrap();
                 if state.fd_to_object_map.contains_key(&msg.fd) {
-                    format!("cinit {},{}", hdr.id, -libc::EALREADY)
+                    unsafe {
+                        libc::close(msg.fd as i32);
+                    }
```

Comment:
When dropping the blob object, the msg.fd should be closed already, seems we should not close fd here.

Reply:
fixed, thanks!